client/src/templates/Challenges/utils/frame.ts

```diff
@@ -273,11 +273,47 @@ const initTestFrame = (frameReady?: () => void) => (frameContext: Context) => {
 const initMainFrame =
   (_: unknown, proxyLogger?: ProxyLogger) => (frameContext: Context) => {
     waitForFrame(frameContext)
-      .then(() => {
+      .then(async () => {
+        const iframeBuild: string = frameContext.build;
+        const parser = new DOMParser();
+        const iframeBuildDoc = parser.parseFromString(iframeBuild, 'text/html');
+        const allLinks = iframeBuildDoc.querySelectorAll(
+          'script[src], link[href]'
+        );
+        let errMsgs = '';
+        let unfetchedLinks = allLinks.length;
+
+        if (!unfetchedLinks) return null;
+
+        for (const url of allLinks) {
+          const urlAddr =
+            url.tagName === 'SCRIPT'
+              ? (url as HTMLScriptElement).src
+              : (url as HTMLLinkElement).href;
+
+          try {
+            const response = await fetch(String(urlAddr));
+            if (!response.ok) {
+              errMsgs = `${errMsgs}\nCannot retrieve ${urlAddr}, link status code: ${response.status}`;
+            } else if (!unfetchedLinks) {
+              continue;
+            }
+            unfetchedLinks -= 1;
+          } catch {
+            errMsgs = `${errMsgs}\n${urlAddr} does not exist. Only files that can be sourced are styles.css, script.js, or remote files`;
```

Comment:
```suggestion
            // TODO: tailor message to the file (e.g. script.js, styles.css etc.)
            return `${url} does not exist. Only files that can be sourced are styles.css, script.js, or remote files`;
```
The message is fine for this PR, but we should address the TODO at some piont!

Reply:
What's the usual deadline for TODOs like this ?

Reply:
Ideally it would be something we address in this PR.